src/assets/src/containers/app.tsx

```diff
@@ -1,52 +1,42 @@
-import { AddQueuePage } from "../components/addQueue";
 import { ManageQueueSettingsPage } from "../components/queueSettings";
+import { SearchPage } from "../components/search";
 import { useGoogleAnalytics } from "../hooks/useGoogleAnalytics";
 
-interface AppProps extends RouteComponentProps {
+interface AppProps {
     globals: Globals;
 }
 
-function App(props: AppProps) {
+export function App(props: AppProps) {
     useGoogleAnalytics(props.globals.ga_tracking_id, props.globals.debug);
+
+    const commonProps: PageProps = {
+        user: props.globals.user,
+        loginUrl: props.globals.login_url,
+        backends: props.globals.backends,
+        defaultBackend: props.globals.default_backend
+    };
+
     return (
-        <Switch>
-            <Route path='/' exact render={p => 
-                <HomePage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/manage' exact render={p => 
-                <ManagePage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/manage/:queue_id' exact render={p => 
-                <QueueManagerPage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} key={(p.match.params as any).queue_id} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/queue/:queue_id' exact render={p => 
-                <QueuePage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} key={(p.match.params as any).queue_id} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/search/' exact render={p =>
-                <SearchPage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/preferences' exact render={p =>
-                <PreferencesPage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/add_queue' exact render={p =>
-                <AddQueuePage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} defaultBackend={props.globals.default_backend} />
-            }/>
-            <Route path='/manage/:queue_id/settings' exact render={p =>
-                <ManageQueueSettingsPage {...p} user={props.globals.user} loginUrl={props.globals.login_url} backends={props.globals.backends} defaultBackend={props.globals.default_backend} />
-            }/>
-        </Switch>
+        <Routes>
```

Comment:
Nice this made the Route patterns a lot cleaner.

Reply:
indeed! https://reactrouter.com/en/main/upgrading/v5#advantages-of-route-element